File: src/JavaScript/easy/string/problem2325.js
/**
 * @param {string} key
 * @param {string} message
 * @return {string}
 */
var decodeMessage = function (key, message) {
  let alphabetLetters = ["a","b","c","d","e","f","g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v","w","x","y","z"];
  let keyArr = Array.from(new Set(key.split("").filter((k) => k != " ")));
  let keyLettersObj = {};
  alphabetLetters.forEach((letter, idx) => {
    keyLettersObj[keyArr[idx]] = letter;
  });
  let decodedMessage = "";
  message.split("").forEach(msg => {
    decodedMessage += msg != " " ? keyLettersObj[msg] : " "
  });
  return decodedMessage
};

let key = "the quick brown fox jumps over the lazy dog",
  message = "vkbs bs t suepuv";

decodeMessage(key, message);

File: src/Python/easy/string/problem2325.py
class Solution:
    def decodeMessage(self, key: str, message: str) -> str:
        alphabetLetters = ["a","b","c","d","e","f","g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v","w","x","y","z"]
        keyList = list(dict.fromkeys(key.replace(' ', '')))
        lettersObj = {}
        decodedMessage = ''
        for idx, k in enumerate(keyList):
            lettersObj[k] = alphabetLetters[idx] 
        for m in message:
            if m == ' ':
                decodedMessage += ' '
            else:
                decodedMessage += lettersObj[m]
        return decodedMessage
                


key = "the quick brown fox jumps over the lazy dog"
message = "vkbs bs t suepuv"

solution = Solution()
solution.decodeMessage(key, message)